tests/expr_templates.cpp

```diff
@@ -54,6 +54,26 @@ TEST(expr_templates, basic_add_lambda)
   }
 }
 
+TEST(expr_templates, subtraction_not_commutative)
+{
+  constexpr int size = 10;
+  mfem::Vector  a(size);
+  mfem::Vector  b(size);
+  mfem::Vector  c(size);
+
+  for (int i = 0; i < size; i++) {
+    a[i] = i * 4 + 1;
+    b[i] = i * i * 3 + 2;
+    c[i] = i * i * i * 7 + 23;
+  }
+
+  mfem::Vector result1 = a + c - b;
```

Comment:
Shouldnt you need a multiply in here instead of just addition and subtraction to test for non-commutativeness (non-commutativity? non-commutatude?) Since we are testing for order of operations seems like you would need a mix of both.